File: src/main/java/com/matag/admin/auth/validators/PasswordValidator.java
package com.matag.admin.auth.validators;

import org.springframework.stereotype.Component;

@Component
public class PasswordValidator implements Validator<String> {
  private static final String PASSWORD_IS_INVALID = "Password is invalid (should be at least 4 characters).";

  @Override
  public void validate(String password) throws ValidationException {
    if (password.length() < 4) {
      throw new ValidationException(PASSWORD_IS_INVALID);
    }
  }
}

File: src/main/java/com/matag/admin/auth/validators/PasswordValidator.kt
package com.matag.admin.auth.validators

import org.springframework.stereotype.Component

@Component
open class PasswordValidator : Validator<String> {
    override fun validate(field: String?) {
        if (field == null || field.length < 4) {
            throw ValidationException(PASSWORD_IS_INVALID)
        }
    }

    companion object {
        private const val PASSWORD_IS_INVALID = "Password is invalid (should be at least 4 characters)."
    }
}
